lib/index.js

```diff
@@ -48,16 +44,14 @@ function auth0GuardianJS(options) {
   self.httpClient = object.get(options, 'dependencies.httpClient',
     httpClient(self.serviceUrl, globalTrackingId));
 
-  self.socketClient = object.get(options, 'dependencies.socketClient');
   self.transport = options.transport || 'socket';
 
-  if (!self.socketClient) {
-    if (options.transport === 'polling') {
-      self.socketClient = pollingClient(self.serviceUrl, { httpClient: self.httpClient });
-    } else {
-      self.socketClient = socketClient(self.serviceUrl);
-    }
-  }
+  self.socketClient = clientFactory({
```

Comment:
`clientFactory.create` will look better, just like `transactionFactory.fromXX`
otherwise the require should have a better name, like:
```
var createClient = require('./utils/client_factory');
```
so the function call is more understandable